build-scripts/build_rule_playbooks.py

```diff
@@ -0,0 +1,335 @@
+        with open(playbook_path, "w") as playbook_file:
+            ssg.yaml.ordered_dump(
+                playbook, playbook_file, default_flow_style=False
+            )
+
+    def open_profile(self, profile_path):
+        """
+        Opens and parses profile at the given profile_path.
+        """
+        if not os.path.isfile(profile_path):
+            raise RuntimeError("'%s' is not a file!\n" % profile_path)
+        profile_id, ext = os.path.splitext(os.path.basename(profile_path))
+        if ext != ".profile":
+            raise RuntimeError(
+                "Encountered file '%s' while looking for profiles, "
+                "extension '%s' is unknown. Skipping..\n"
+                % (profile_path, ext)
+            )
+
+        profile = ssg.build_yaml.Profile.from_yaml(profile_path, self.env_yaml)
+        if not profile:
+            raise RuntimeError("Could not parse profile %s.\n" % profile_path)
+        return profile
+
+    def create_playbooks_for_all_rules(self, profile, variables):
+        """
+        Creates a Playbook for each rule selected in a profile from tasks
+        extracted from snippets. Created Playbooks are parametrized by
+        variables according to profile selection. Playbooks are written into
+        a new subdirectory in output_dir.
+        """
+        profile_rules, profile_refines = self.get_profile_selections(profile)
+
+        profile_playbooks_dir = os.path.join(self.output_dir, profile.id_)
+        os.makedirs(profile_playbooks_dir)
+
+        for snippet in os.listdir(self.input_dir):
+            snippet_path = os.path.join(self.input_dir, snippet)
+            rule_id, ext = os.path.splitext(os.path.basename(snippet_path))
+            if ext != ".yml":
+                sys.stderr.write(
+                    "Found file '%s' while looking for Ansible snippets, "
+                    "extension '%s' is unknown. Skipping...\n"
+                    % (snippet_path, ext)
+                )
+            if rule_id in profile_rules:
+                self.create_playbook(
+                    snippet_path, rule_id, variables,
+                    profile_refines, profile_playbooks_dir
+                )
+
+    def create_playbook_for_single_rule(self, profile, rule_id, variables):
+        """
+        Creates a Playbook for given rule specified by a rule_id. Created
+        Playbooks are parametrized by variables according to profile selection.
+        Playbooks are written into a new subdirectory in output_dir.
+        """
+        profile_rules, profile_refines = self.get_profile_selections(profile)
+        profile_playbooks_dir = os.path.join(self.output_dir, profile.id_)
+        os.makedirs(profile_playbooks_dir)
+        snippet_path = os.path.join(self.input_dir, rule_id + ".yml")
+        if rule_id in profile_rules:
+            self.create_playbook(
+                snippet_path, rule_id, variables,
+                profile_refines, profile_playbooks_dir
+            )
```

Comment:
Please, add an error message if specified `rule_id` is not part of the profile.

Reply:
Done.